pkgs/applications/misc/noice/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchgit, ncurses, conf ? null }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "noice-${version}";
+  version = "0.6";
+
+  src = fetchgit {
+    url = "git://git.2f30.org/noice.git";
+    rev = "refs/tags/v${version}";
+    sha256 = "03rwglcy47fh6rb630vws10m95bxpcfv47nxrlws2li2ljam8prw";
+  };
+
+  configFile = optionalString (conf!=null) (builtins.toFile "config.def.h" conf);
+  preBuild = optionalString (conf!=null) "cp ${configFile} config.def.h";
```

Comment:
I thought so too but somehow a lot of other packages are doing it the same way. Take [abduco](https://github.com/NixOS/nixpkgs/blob/56904d7c423f2b13b37fbd29f39bbb4b52bc7824/pkgs/tools/misc/abduco/default.nix) or [st](https://github.com/NixOS/nixpkgs/blob/7d97051f7212a3d6c5e1209bdc8e6acef3bcfe18/pkgs/applications/misc/st/default.nix) for example. Any good reason for this?